CCDP-1147: Documentation updates

mhelf-dolby opened the pull request.

kubaau commented on README.md:
```diff
@@ -96,19 +96,19 @@ Example:
 
 ### Mute Input
-Mutes audio input. Has no effect unless connected.
+Mutes audio input. The method has no effect unless an audio input device is connected.
```
This (and several statements below) is incorrect, but also means that the initial comment was unclear in the first place. These methods have no effect unless connected to a conference. It's a little unusual but the limimitation is, AFAIK, enforced by the C++ SDK.

kubaau commented on README.md:
```diff
@@ -78,7 +78,7 @@ Example:
 
 ### Set Token Using Key and Secret
-Similar to [Set Token](#set-token) except it takes an app key and secret as parameters and automatically generates a token. **Do not use this function in production and do not allow your app key and secret to leak.**
+For convenience during early development and prototyping, this function is provided to acquire a client access token directly from within the application. However, please note **we do not recommend** using this mechanism in the production software for [security best practices](https://docs.dolby.io/communications-apis/docs/guides-client-authentication). App secret needs to be protected and not in ...
```
I would keep the first sentence of the original description as well, at least the part about this function being similar to Set Token as it has the same effect in terms of initializing the plugin.

kubaau commented on README.md:
```diff
@@ -16,59 +16,59 @@ Note: If you want to use the plugin on macOS, see our [advice](#macos-advice).
 ## Functions
 ### Set Token
-Initializes or refreshes the client access token. Takes the token as parameter. Initializes the plugin unless already initialized. Triggers [On Initialized](#on-initialized) if initialization was successful.
+Initializes or refreshes the client access token. The function takes the token as a parameter and initializes the plugin unless already initialized. A successful initialization triggers the [On Initialized](#on-initialized) event.
```
"A successful initialization" doesn't sound quite right to me. The "a" article is unnecessary here if we want to keep this wording.

kubaau commented on README.md:
```diff
@@ -16,59 +16,59 @@ Note: If you want to use the plugin on macOS, see our [advice](#macos-advice).
 
 ### On Connected
-Successfully connected to conference. Triggered by the [Connect](#connect) function. Provides the ID of the local participant in its argument.
+Triggered when a client is successfully connected to a conference after calling the [Connect](#connect) function. The event provides the ID of the local participant in its argument.
```
"A client successfully connected to a conference" sounds like it applies to any client in any ongoing conference. There can only be at most one client and one conference in the plugin at any given moment, so I think "the" is correct article in both cases. The same applies to the two following events.

kubaau reviewed (changes requested):
Approved except for the descriptions of the mute/unmute functions.

mhelf-dolby commented:
I updated the Readme file and added the same descriptions to the DolbyIO.h file. If I should limit of the number of characters per line in this file, let me know.

kubaau commented:
> I updated the Readme file and added the same descriptions to the DolbyIO.h file. If I should limit of the number of characters per line in this file, let me know.

Don't worry about it, I can automatically format this file after we merge if there are any "issues".

kubaau commented on DolbyIO/Source/DolbyIOModule/Public/DolbyIO.h:
```diff
@@ -28,62 +28,89 @@ class DOLBYIOMODULE_API UDolbyIO : public UGameInstanceSubsystem, public DolbyIO
+	 *
+	 * You may use the [Set Token Using Key and Secret](#set-token-using-key-and-secret) function instead for convenience during onboarding.
+	 *
+	 * Example:
+	 * ![example](./../../../../Images/on_token_needed.PNG)
```
These examples probably won't be visible in Unreal Editor. I do think there's value in trying to make it possible to automatically generate documentation based on this file, so that it becomes the ultimate source of information on the plugin and therefore putting the example image links inside it makes some sense, but I'm not sure about it and definitely not about using relative paths like this. Using relative paths with look silly to someone just looking at the code and will simply not work when we put the plugin on Epic Marketplace. The plugin's code will be there, but the images will not, so they would at least have to be linked to specific files on the Internet.
